R/utils-code_dependency.R

```diff
@@ -0,0 +1,336 @@
+  symbol_effects_names <-
+    unlist(
+      lapply(
+        calls_pd,
+        function(x) {
+          com_cond <-
+            x$token == "COMMENT" & grepl("@linksto", x$text) & grepl(paste0("[\\s]*", object, "[\\s$]*"), x$text)
+
+          # Make sure comment id is not the highest id in the item.
+          # For calls like 'options(prompt = ">") # @linksto ADLB',
+          # 'options(prompt = ">")' is put in a one item
+          # and '# @linksto ADLB' is the first element of the next item.
+          # This is tackled in B.
+
+
+          if (!com_cond[1] & sum(com_cond) > 0) {
+            # A.
+            x[x$token == "SYMBOL", "text"]
+          } else if (com_cond[1] & sum(com_cond[-1]) > 0) {
+            # B.
+            x <- x[-1, ]
+            x[x$token == "SYMBOL", "text"]
+          }
+        }
+      )
+    )
+
+  commented_calls <- vapply(
+    calls_pd,
+    function(x) any(x$token == "COMMENT" & grepl("@linksto", x$text)),
+    logical(1)
+  )
+
+  symbol_effects_lines <-
+    unlist(
+      lapply(
+        symbol_effects_names,
+        function(x) {
+          code <- return_code(x, occur = occur, cooccur = cooccur, eff = eff)
+          if (is.null(code)) {
+            # NOT SURE IF BELOW IS NEEDED ANYMORE ONCE WE MOVE TO SYMBOLS
+            # Extract lines for objects that were used, but never created.
+            # Some objects like 'iris' or 'mtcars' are pre-assigned in the session.
+            # Below is just used for comments with @linksto.
+            # if (!object %in% names(occur)) {
+            intersect(which(detect_symbol(x, calls_pd)), which(commented_calls))
+            # }
+          } else {
+            code
+          }
+        }
+      )
+    )
+
+  # When commet_id is the highest id in the item - take previous item.
+  side_effects_names <-
+    unlist(
+      lapply(
+        calls_pd,
+        function(x) {
+          com_cond <-
+            x$token == "COMMENT" & grepl("@linksto", x$text) & grepl(paste0("[\\s]*", object, "[\\s$]*"), x$text)
+
+          # Work out the situation when comment id is the highest id in the item.
+          # For calls like 'options(prompt = ">") # @linksto ADLB',
+          # 'options(prompt = ">")' is put in a one item
+          # and '# @linksto ADLB' is the first element of the next item.
+
+          com_cond[1]
+        }
+      )
+    )
+
+  side_effects_lines <- which(side_effects_names) - 1
+
+  sort(unique(c(symbol_effects_lines, side_effects_lines)))
+}
+
+#' Return the lines of code (with side-effects) needed to reproduce the object.
+#' @return `character` vector of elements of `parsed_code` calls that were required to build the side-effects and
+#' influencing objects having and impact on the `object`
+#'
+#' @param qenv `qenv` object
+#' @param names `character` with object names
+#' @keywords internal
+get_code_dependency <- function(qenv, names) {
+  parsed_code <- parse(text = as.character(qenv@code), keep.source = TRUE)
```

Comment:
get_code_dependency can work fine without `qenv` being passed. 
 
```suggestion
#' @param qenv `character` object
#' @param names `character` with object names
#' @keywords internal
get_code_dependency <- function(code, names) {
  checkmate::assert_string(code)
  checkmate::assert_character(names)
  parsed_code <- parse(text = as.character(code), keep.source = TRUE)
```

Reply:
TRUE, it will also require the change on get_code(). Also code can be a character(n) so it should be checkmate::assert_character()

Reply:
82830e446d298d0f6c37d2e919e2d44417210f8d